### ISSUE TITLE
Offset race condition / cache flush on redeploy ? 

### ISSUE THREAD

**Issue by dannygoldstein** [skyportal/skyportal#1345]
On fritz @ 9ee5f68,

If I go to https://fritz.science/run/8, 

The run will send an offset request for each target on the run (~32 requests) to generate the composite starlist. 

On the backend, this produces, 

```
ERROR:tornado.application:Uncaught exception GET /api/sources/ZTF20acfvqjl/offsets?facility=Keck (127.0.0.1)
HTTPServerRequest(protocol='http', host='fritz.science', method='GET', uri='/api/sources/ZTF20acfvqjl/offsets?facility=Keck', version='HTTP/1.0', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/skyportal_env/lib/python3.8/site-packages/tornado/web.py", line 1697, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "/skyportal/baselayer/app/access.py", line 32, in wrapper
    return tornado.web.authenticated(method)(self, *args, **kwargs)
  File "/skyportal_env/lib/python3.8/site-packages/tornado/web.py", line 3174, in wrapper
    return method(self, *args, **kwargs)
  File "/skyportal/skyportal/handlers/api/source.py", line 924, in get
    ) = get_nearby_offset_stars(
  File "/skyportal/skyportal/utils/offset.py", line 43, in decorated_function
    return func(*args, **kwargs)
  File "/skyportal/skyportal/utils/offset.py", line 43, in decorated_function
    return func(*args, **kwargs)
  File "/skyportal/skyportal/utils/offset.py", line 443, in get_nearby_offset_stars
    ztfcatalog = get_ztfcatalog(source_ra, source_dec)
  File "/skyportal/skyportal/utils/offset.py", line 212, in get_ztfcatalog
    cache[catname] = buf.read()
  File "/skyportal/skyportal/utils/cache.py", line 72, in __setitem__
    self.clean_cache()
  File "/skyportal/skyportal/utils/cache.py", line 76, in clean_cache
    cached_files = [
  File "/skyportal/skyportal/utils/cache.py", line 77, in <listcomp>
    (f.stat().st_mtime, f.absolute()) for f in self._cache_dir.glob('*')
  File "/usr/lib/python3.8/pathlib.py", line 1194, in stat
    return self._accessor.stat(self)
FileNotFoundError: [Errno 2] No such file or directory: 'cache/finder_cat/aad1510ab09b256f8e21ad8b200e0f64'
```

which brings the whole site down (502s). 

Could this be due to the cache being flushed on redeploy, or a race condition in which multiple requests require the same cache resource and whicheverone finishes first deletes the resource needed by the other?

**Comment by dannygoldstein** [comment]
grepping for the resource in question, I see


```
(base) Dannys-MBP:skyportal danny$ kubectl logs fritz-1601421112-web-df95d57fc-rp6rj | grep aad1510ab09b256f8e21ad8b200e0f64
[21:14:12 cache] save [ztf_000748_zg_c15_q1_refpsfcat.fits] to [aad1510ab09b256f8e21ad8b200e0f64]
[21:14:38 cache] cleanup [aad1510ab09b256f8e21ad8b200e0f64]
[21:14:41 cache] save [ztf_000748_zg_c15_q1_refpsfcat.fits] to [aad1510ab09b256f8e21ad8b200e0f64]
[21:15:12 cache] cleanup [aad1510ab09b256f8e21ad8b200e0f64]
[21:15:37 cache] save [ztf_000748_zg_c15_q1_refpsfcat.fits] to [aad1510ab09b256f8e21ad8b200e0f64]
FileNotFoundError: [Errno 2] No such file or directory: 'cache/finder_cat/aad1510ab09b256f8e21ad8b200e0f64'
[21:19:11 basehandler] Error response returned by [/api/sources/ZTF20acfvqjl/offsets]: [[Errno 2] No such file or directory: 'cache/finder_cat/aad1510ab09b256f8e21ad8b200e0f64']
[21:19:11 cache] cleanup [aad1510ab09b256f8e21ad8b200e0f64]
[21:29:10 cache] save [ztf_000748_zg_c15_q1_refpsfcat.fits] to [aad1510ab09b256f8e21ad8b200e0f64]
[21:34:29 cache] cleanup [aad1510ab09b256f8e21ad8b200e0f64]
[21:34:47 cache] save [ztf_000748_zg_c15_q1_refpsfcat.fits] to [aad1510ab09b256f8e21ad8b200e0f64]
[21:35:16 cache] cleanup [aad1510ab09b256f8e21ad8b200e0f64]
```

**Comment by profjsb** [comment]
The number of cached items is ridiculously small given how small the data is (compared to say the DB). There's probably some deep logic to implement in terms of locks but I'll bump the cache sizes to see if that fixes things in prod.

**Comment by dannygoldstein** [comment]
#1346 has solved the problem in production, so removing the high priority pin, but leaving this issue open in case we want to develop a lock based solution for concurrent access of the same resource by multiple web workers (I think that was the problem here).

**Comment by stefanv** [comment]
I don't think this is a race condition, necessarily.  8 requests will block all serving nodes long enough that the site will present as down (it's not, but nginx has no way of knowing that).  Increasing the cache size won't fix the issue unless there are repeated queries, but #1350 should do the trick.

**Comment by dannygoldstein** [comment]
Fixed by #1350
